### ISSUE TITLE
Enforce name consistency with `NewType` call

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6829]
Pyright doesn't currently enforce naming consistency with a `NewType` call. Mypy does.

The following should be flagged as an error.

```python
MyInt = NewType("NotMyInt", int)
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.344, which I just published. It will also be included in a future release of pylance.
